verifiable-db/src/row_tree/leaf.rs

```diff
@@ -20,31 +20,32 @@ use recursion_framework::{
 };
 use serde::{Deserialize, Serialize};
 
-use crate::cells_tree;
+use crate::cells_tree::{self, circuit_accumulate_proof_digest, Cell, CellWire};
 
-use super::{public_inputs::PublicInputs, IndexTuple, IndexTupleWire};
-use derive_more::{Constructor, Deref, From};
+use super::public_inputs::PublicInputs;
 
 // new type to implement the circuit logic on each differently
 // deref to access directly the same members - read only so it's ok
-#[derive(Clone, Debug, Deref, From, Constructor)]
-pub struct LeafCircuit(IndexTuple);
+#[derive(Clone, Debug, From, Into)]
+pub struct LeafCircuit(Cell);
 
-#[derive(Clone, Serialize, Deserialize, Deref, From)]
-pub(crate) struct LeafWires(IndexTupleWire);
+#[derive(Clone, Serialize, Deserialize, From, Into)]
+pub(crate) struct LeafWires(CellWire);
 
 impl LeafCircuit {
     pub(crate) fn build(b: &mut CircuitBuilder<F, D>, cells_pis: &[Target]) -> LeafWires {
         let cells_pis = cells_tree::PublicInputs::from_slice(cells_pis);
         // D(index_id||pack_u32(index_value)
-        let tuple = IndexTupleWire::new(b);
-        let d1 = tuple.digest(b);
-        // D(proof.DC + D(index_id||pack_u32(index_value)))
-        let cells_digest = cells_pis.digest_target();
-        let input_digest = b.curve_add(cells_digest, d1);
-        let row_digest = b
-            .map_to_curve_point(&input_digest.to_targets())
-            .to_targets();
+        let tuple = CellWire::new(b);
+        // set the right digest depending on the multiplier and accumulate the ones from the public
+        // inputs of the cell root proof
+        let (digest_ind, digest_mult) = tuple.split_and_accumulate_digest(b, &cells_pis);
+        // final_digest = HashToInt(mul_digest) * D(ind_digest)
+        // NOTE This additional digest is necessary since the individual digest is supposed to be a
+        // full row, that is how it is extracted from MPT
+        let digest_ind = b.map_to_curve_point(&digest_ind.to_targets());
+        let final_digest =
+            circuit_hashed_scalar_mul(b, digest_mult.to_targets(), digest_ind).to_targets();
```

Comment:
If we always compute the `final_digest` in this way, won't we have a mismatch in case we don't merge simple and mapping variables with the merge circuit? Because in this case the digest would still be without any multiplier, i.e., the old format, no? Shouldn't we have a further flag that specifies whether to compute `final_digest` in this way or by simply taking `digest_ind`?